### PR TITLE
fix(users pagination): fix users pagination

### PR DESCRIPTION
**Pull request by Thuku**
## What does this PR do?
 -  it persists paginated users in the store and avoids fetching of users if they are already in the store

## Description of Task to be completed?
- For stored data, on clicking previous page number, the API should not be invoked again and data is displayed very fast.
## How should this be manually tested?
 - pull this branch and click on users
 - Load data on several pages and every time you click on the loaded page data will be displayed instantly
## What are the relevant pivotal tracker stories?
 - https://www.pivotaltracker.com/story/show/160395201
## Important notes
 - none
## Packages installed
 - none
## Deployment note
 - none

### CODE REVIEW

**Comment by sachungo** [comment]
@Thuku how far is the test? Also, I have gone through your changes again and realized that `handleRowChange` doesn't use `retrieveUsers` method. This means that it never considers the cutoff value

**Comment by Thuku** [comment]
> @Thuku how far is the test? Also, I have gone through your changes again and realized that `handleRowChange` doesn't use `retrieveUsers` method. This means that it never considers the cutoff value

Tests are good let me check `retrieveUsers`

File: src/_components/User/UserContainer.jsx [hunk 1]
```diff
@@ -5,50 +5,65 @@ import PropTypes from 'prop-types';
...
+    if (usersEmpty) {
+      this.props.loadUsers(this.props.activePage, this.state.limit);
     }
-    return true;
   }
 
   handleRowChange = (e, data) => {
     this.setState({ limit: data.value });
-    this.props.loadUsers(this.state.activePage, data.value);
+    this.retrieveUsers(this.state.activePage, data.value);
```
Comment (sachungo) [inline src/_components/User/UserContainer.jsx]: Shouldn't the parameter be `this.props.activePage`?

File: src/_reducers/users.reducer.js [hunk 2]
```diff
@@ -47,6 +52,16 @@ export default (state = initialState.usersList, action) => {
         ...state,
         errorMessage: action.payload
       };
+    case RESET_USERS:
+      return {
+        ...state,
+        assetsList: {}
```
Comment (sachungo) [inline src/_reducers/users.reducer.js]: `assetsList`? I thought this was users' reducer

File: src/_actions/users.actions.js [hunk 3]
```diff
@@ -55,17 +64,30 @@ const loadAssetAssigneeSuccess = users => ({
...
+      dispatch({
+        type: CREATE_SECURITY_USER_FAILURE,
+        payload: error.message
+      });
+      dispatch(updateToastMessageContent('Could not save Security User', 'error'));
     });
-    dispatch(updateToastMessageContent('Could not save Security User', 'error'));
-  });
+
+export const setActivePage = page => dispatch => dispatch(setActivePageSuccess(page));
```
Comment (sachungo) [inline src/_actions/users.actions.js]: No need for this.  Just export `setActivePageSuccess` and use it wherever `setActivePage` will be used. Only use `dispatch` in action creators when using thunk middleware to make API calls

File: src/_reducers/initialState.js [hunk 4]
```diff
@@ -67,13 +67,15 @@ export default {
...
+    users: {},
     assetAsigneeUsers: [],
     usersCount: 0,
     errorMessage: '',
     hasError: false,
     isLoading: false,
-    securityUser: {}
+    securityUser: {},
+    activePage: 1
+
```
Comment (sachungo) [inline src/_reducers/initialState.js]: Delete this empty line

**Pull request merged**